projects/cashmere/src/lib/pipes/file-size/file-size.pipe.ts

```diff
@@ -0,0 +1,35 @@
+import { Pipe, PipeTransform } from "@angular/core";
+
+// Convert bytes into largest possible unit.
+// Credit to https://gist.github.com/JonCatmull/ecdf9441aaa37336d9ae2c7f9cb7289a
+@Pipe({ "name": "fileSize" })
+export class FileSizePipe implements PipeTransform {
+
+    private units: string[] = [
+        "bytes",
+        "KB",
+        "MB",
+        "GB",
+        "TB",
+        "PB"
+    ];
+
+    public transform(bytes: number = 0, precision: number = 2): string {
+        if (isNaN(parseFloat(String(bytes))) || !isFinite(bytes)) {
```

Comment:
I don't see any tests covering this condition.

Reply:
Also, I think this would be a cleaner way to do this check:

```ts
        bytes = +bytes;
        if (isNaN(bytes) || !isFinite(bytes)) {
```

Reply:
implemented this code